### ISSUE TITLE
SARIF report should only include active rules for RuleDescriptors

### ISSUE THREAD

**Issue by chao2zhang** [detekt/detekt#3554]
## Expected Behavior
SARIF report should only include active rules

## Observed Behavior
SARIF report includes all rules on the classpath  

## Context
This should help reduce the output size as well as confusion to readers of SARIF.

## Your Environment
* Version of detekt used: 1.16.0

**Comment by BraisGabin** [comment]
Could you add some reference to the documentation?

**Comment by BraisGabin** [comment]
@chao2zhang was this issue solved by #3556? What should be done here to close this?

**Comment by chao2zhang** [comment]
This issue requires only printing resultDescriptors of rules that are active/enabled. I can take another look this week.

**Comment by TWiStErRob** [comment]
Hmm, interesting, I just wanted to ask for the opposite. Don't understand why a sarif file needs to contain all rules that exist inside Detekt. For me, an empty report means there were no issues.
detekt.txt 0 bytes
detekt.xml 79 bytes (`<checkstyle />`)
detekt.html 2200 bytes (CSS + metrics)
detekt.sarif 76k (all rule definitions)

Now, imagine I have a project with 200 modules, each running Detekt. That's 15MB per CI build. Now let's have 30 developers, pushing 5-10 times a day, that's 3GBs worth of "empty" reports a day in artifacts.

So, would it be possible to only include rules that are Detekted in the report, or is that going against SARIF?

**Comment by chao2zhang** [comment]
I believe it should
- Definitely not include all rules because of the waste of space
- Include enabled rules because those were active in that detekt run

If the size is a concern for you to adopt SARIF format, we can definitely add some flexibility of configuring how to generate lightweight SARIF or not.

**Comment by TWiStErRob** [comment]
It's not an issue yet (we're not actively using Sarif yet), and thinking more about it, I think it might not be in the future, because we can easily remove a 200x multiplier:
```kotlin
tasks.register<io.gitlab.arturbosch.detekt.report.ReportMergeTask>("detektReportMergeSarif") {
	output.set(rootProject.buildDir.resolve("reports/detekt/merge.sarif"))
}
allprojects { project ->
	val detektReportMergeSarif = rootProject.tasks.named<ReportMergeTask>("detektReportMergeSarif")
	project.tasks.withType<Detekt> {
		finalizedBy(detektReportMergeSarif)
		detektReportMergeSarif.configure { input.from(this@withType.sarifReportFile) }
	}
}
```

So only do this if it's worth the effort of supporting this. (https://xyproblem.info 😅).

**Comment by TWiStErRob** [comment]
Still an issue, we just ran out of CI space because of another report :) Still didn't get to use type resolution and per-module tasks, but I think this would be very useful when we do.

**Comment by TWiStErRob** [comment]
This really wants to close itself 🫣.

**Comment by BraisGabin** [comment]
It shouldn't be stale anymore with the "help wanted" flag. I think this issue is clear so anyone could take care of it. But this is not an easy one. We don't have this information at that level.

**Comment by BraisGabin** [comment]
This was done already at #7866
